src/backend/utils/text-parser.js

```diff
@@ -1,20 +1,33 @@
-const puppeteer = require('puppeteer');
+const { Cluster } = require('puppeteer-cluster');
 
-module.exports = async function(htmlFragment) {
-  let browser;
-  try {
-    browser = await puppeteer.launch();
-    const page = await browser.newPage();
+let isInitialized = false;
+let cluster;
 
-    await page.setContent(htmlFragment, {
+module.exports.parse = async function(htmlFragment) {
```

Comment:
Let's not bother with `.parse` here.  Let's just expose the function directly.

Reply:
will I still be able to call `textParser.close()` if I do this?

Reply:
Ah, I see.  OK, leave it like this.